lib/plugins/aws/deploy/lib/splitStack.js

```diff
@@ -0,0 +1,212 @@
+'use strict';
+
+const path = require('path');
+const _ = require('lodash');
+const BbPromise = require('bluebird');
+const traverse = require('traverse');
+const DepGraph = require('dependency-graph').DepGraph;
+
+module.exports = {
+  splitStack() {
+    if (!this.serverless.service.provider.useStackSplitting) return BbPromise.resolve();
+
+    return BbPromise.bind(this)
+      .then(this.createDepGraph)
+      .then(this.generateStacks)
+      .then(this.uploadStackFiles)
+      .then(this.updateCompiledCloudFormationTemplate)
+      .then(this.writeStacksToDisk);
```

Comment:
Thanks for the feedback @dougmoscrop 

Right now it's implemented this way so that we have all the URLs (e.g. to the function zips) in place.